scipy/signal/_short_time_fft.py

```diff
@@ -73,6 +73,153 @@ class method ``ShortTimeFFT.from_dual()``.
     return win / DD
 
 
+def closest_STFT_dual_window(win: np.ndarray, hop: int, desired_dual: np.ndarray, *,
```

Comment:
So it looks like the vast majority of uses in examples and tests are of the form:
```
closest_STFT_dual_window(win, hop, np.ones(len(win))
```
with a few select others others being
```
closest_STFT_dual_window(np.ones(len(desired_dual)), hop, desired_dual)
```
i.e., one of `win` or (much more frequently) `desired_dual` is `np.ones(len(the_other))`.

Is it worth making this explicit by changing the signature to:
```
desired_dual : np.ndarray | None = None
```
? So given a `win` and `hop` you can more easily get a dual that is closest to a `rect` (which seems like the by-far most common case)?

Reply:
I agree with you: Rectangular windows a probably the most common use case. Done in commit 0142714d7f.